packages/backend/src/peripherals/starkex/StarkexClient.ts

```diff
@@ -0,0 +1,76 @@
+import { getErrorMessage, HttpClient, Logger } from '@l2beat/common'
+import { StarkexProduct } from '@l2beat/config'
+import { json } from '@l2beat/types'
+
+import { parseStarkexApiResponse } from './parseStarkexApiResponse'
+
+export class StarkexClient {
+  constructor(
+    private starkexApiUrl: string,
+    private starkexApiKey: string,
+    private httpClient: HttpClient,
+    private logger: Logger,
+  ) {
+    this.logger = this.logger.for(this)
+  }
+  // Rate limit?
+
+  // TODO: create BaseClient for this and EtherscanClient to avoid duplication
```

Comment:
Do they have this much in common?

Reply:
I was mostly thinking of ErrorHandling, maybe it could be done in HttpClient?